src/dataflow/src/render/reduce.rs

```diff
@@ -407,15 +798,21 @@ where
         distinct,
     } = aggr.clone();
 
+    // Extract the value we were asked to aggregate over.
     let mut partial = if !prepend_key {
         let mut packer = RowPacker::new();
-        ok_input.map(move |(key, row)| {
+        input.map(move |(key, row)| {
             let value = row.iter().nth(index).unwrap();
             packer.push(value);
             (key, packer.finish_and_reuse())
         })
     } else {
-        ok_input
+        // If the arrangement produced by this function is going to be exported
+        // for reuse that implies that theres only a single aggregation in the
+        // whole reduce, and only one value in the values row. Let's make sure
+        // we're not trying to aggregate over anything else.
+        assert!(index == 0);
+        input
```

Comment:
Is this only true because of our current implementation strategy? Like, if we fused together basic aggregations, this would stop being true. Is this assert testing our current beliefs about the code, or ensuring a correctness property?

Reply:
its testing a belief about the code. specifically these lines

```
 854                 let iter = source.iter().flat_map(|(v, w)| {
 855                     std::iter::repeat(v.iter().next().unwrap()).take(*w as usize)
 856                 });
 ```
 
 only look at the 0th element afaict, so if prepend keys is true and we are looking for a different index we might not crash but we wont get what we want afaict

Reply:
i downgraded this to a log for now